utils/filesystem.go

```diff
@@ -113,63 +113,69 @@ func PingHealth(healthEndpoint string) bool {
 }
 
 //GetZipURL from github api /repos/:owner/:repo/:archive_format/:ref
-func GetZipURL(owner, repo, branch string) string {
+func GetZipURL(owner, repo, branch string) (string, error) {
 	client := github.NewClient(nil)
 
 	opt := &github.RepositoryContentGetOptions{Ref: branch}
 
 	URL, _, err := client.Repositories.GetArchiveLink(context.Background(), owner, repo, "zipball", opt)
 	if err != nil {
-		log.Fatal(err)
+		return "", err
 	}
 	url := URL.String()
-	fmt.Println("Repository archive link - ", URL)
-	return url
+	return url, nil
 }
 
-//DownloadFile into /temp dir using the git archive link
-func DownloadFile(zipFileName, url string) error {
-
+//DownloadFile from URL to file destination
+func DownloadFile(URL, destination string) error {
 	// Get the data
-	resp, err := http.Get(url)
-	errors.CheckErr(err, 400, "")
+	resp, err := http.Get(URL)
+	if err != nil {
+		return err
+	}
 	defer resp.Body.Close()
 
 	// Create the file
-	file, err := os.Create(zipFileName)
-	errors.CheckErr(err, 401, "")
+	file, err := os.Create(destination)
+	if err != nil {
+		return err
+	}
 	defer file.Close()
 
 	// Write body to file
 	_, err = io.Copy(file, resp.Body)
-	fmt.Println(zipFileName)
+	log.Printf("Downloaded file: from '%s' to '%s'\n", URL, destination)
```

Comment:
Again, i dont think this is necessary to change `fmt` to `log`. We dont need to print to stderr here. `fmt.Printf()` should be enough here.